src/foam/comics/v2/BasicEnabledActionsAuth.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.comics.v2',
+  name: 'BasicEnabledActionsAuth',
+  implements: [
+    'foam.comics.v2.EnabledActionsAuth'
+  ],
+  javaImports: [
+    'foam.util.SafetyUtil'
+  ],
+
+  documentation: `
+    String arrays of permissions that will be checked on each CRUD action through availablePermissions
+  `,
+
+  properties: [
+    {
+      class: 'String',
+      name: 'modelName'
+    },
+  ],
+
+  methods: [
+    {
+      name: 'permissionFactory',
+      type: 'String',
+      args: [
+        {
+          name: 'operation',
+          type: 'foam.nanos.ruler.Operations'
+        },
+        {
+          name: 'obj',
+          type: 'FObject'
+        }
+      ],
+      javaCode: `
+        String outputString = getModelName();
+
+        if ( SafetyUtil.equals(operation, foam.nanos.ruler.Operations.CREATE ) ){
+          outputString += ".create";
+        } else if ( SafetyUtil.equals(operation, foam.nanos.ruler.Operations.UPDATE) ) {
+          outputString += ".update." + obj.getProperty("id");
+        } else if ( SafetyUtil.equals(operation, foam.nanos.ruler.Operations.REMOVE) ) {
+          outputString += ".remove." + obj.getProperty("id");
+        } else {
+          throw new RuntimeException("Submitted an invalid operation");
```

Comment:
```suggestion
          throw new RuntimeException(new UnsupportedOperationException(operation.getLabe());
```

Reply:
Gunna fix to operation.getLabel() and push

Reply:
Done.